### PR TITLE
Prepare product_import_spec.rb for BUU as default

### LINKED ISSUE #12627: [BUU] Activate admin_style_v3 for dev, test and new servers
When the new Backoffice UI Uplift design (`admin_style_v3`) is the primary design, we need ensure we are using it while developing new changes:

* dev environments
* tests (local and CI)
* ~~staging~~ (these can be managed manually for now)
* new servers

**Steps:**

1. Prepare tests
   * [x] https://github.com/openfoodfoundation/openfoodnetwork/pull/11645
   * [x] https://github.com/openfoodfoundation/openfoodnetwork/pull/12649
   * [ ] spec/system/admin/product_import_spec.rb:100:213:679:720 - https://github.com/openfoodfoundation/openfoodnetwork/pull/12668
   * [x] spec/system/admin/products_spec.rb:283 - https://github.com/openfoodfoundation/openfoodnetwork/pull/12652
   * [x] spec/system/admin/subscriptions/smoke_tests_spec.rb:240 - https://github.com/openfoodfoundation/openfoodnetwork/pull/12652
   * Consider improving specs as per [review comments](https://github.com/openfoodfoundation/openfoodnetwork/pull/11645#pullrequestreview-2152471490)
- [ ] Make `admin_style_v3` [active by default](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/lib/open_food_network/feature_toggle.rb)
   - This will enable for dev, test and new servers (any environment that doesn't already have a saved setting for `admin_style_v3`)
   - So we can remove the `Flipper.enable` in `base_spec_helper`

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- Closes #12627

* Prepares `./spec/system/admin/product_import_spec.rb` for BUU.
* Addresses review comment [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/11645#discussion_r1673154618).
* Make `admin_style_v3` active by default for new environments
* Migrate `admin_style_v3` to be active on all dev environments

For reviewing: suggest viewing overall diff.

#### What should we test?

Feature `admin_style_v3` should be activated by default in the:
- dev environment (if non-existent)
- test environment (build should be green)

Staging and production environments:
- **should not** be affected by this PR, if the feature is activated in some way :warning: 
- **should** affected by this PR, if the feature is fully disabled, and missing from the list of features in the Flipper settings screen :warning: 

Test instructions:

1)

- Take a staging server and delete the `admin_style_v3` feature
- Stage the PR
-> Verify that the feature is fully enabled

2)

- Take a staging server make sure the `admin_style_v3` feature exists
- Set the feature to be partially enabled (for example, for 25% of the users, or for a specific user)
- Stage the PR
-> Verify that there are no changes on the `admin_style_v3` feature settings (set on the second step)

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: db/migrate/20240715103415_enable_admin_style_v3_by_default.rb
Comment (filipefurtad0) [inline db/migrate/20240715103415_enable_admin_style_v3_by_default.rb:3]: I'm not sure why this is not working :disappointed:  Any ideas @dacook?

Comment (dacook) [reply db/migrate/20240715103415_enable_admin_style_v3_by_default.rb:3]: Hi Filipe, only changes to the database schema will affect the test environment, so this won't affect it. 

Instead, we can make it enabled by default using a different method. I'll add a note on the issue to explain (I had to check it to work out how it works mysefl!)

Comment (dacook) [reply db/migrate/20240715103415_enable_admin_style_v3_by_default.rb:3]: However, enabling it by default won't change any existing Dev environments, so this migration would be helpful. But we need to make sure it only applies to dev envs (we don't want to enable on prod just yet!). You could add a condition inside `up` that checks `if Rails.env.development?`

File: spec/system/admin/product_import_spec.rb
Comment (dacook) [inline spec/system/admin/product_import_spec.rb:689]: Instead of asserting backend values, we can use `ProductsHelper#row_containing_name` to find the table row containing a field in column labelled "name" with the text:
```suggestion
      within row_containing_name("Carrots") do
```

(You'll need to include `ProductsHelper` at the top of the file)

File: spec/system/admin/product_import_spec.rb
Comment (dacook) [inline spec/system/admin/product_import_spec.rb:693]: Also here we can use the column labels to refer:

```suggestion
        expect(page).to have_input("Name", text: "Carrots") # (This is now already asserted by row_containing_name above)
        expect(page).to have_input("Unit", text: "1 lb")
```

File: spec/system/admin/product_import_spec.rb
Comment (filipefurtad0) [reply spec/system/admin/product_import_spec.rb:689]: I'm not sure I can make it work with `row_containing_name`, as it finds the first row which contains the name "Carrots" (the product row - `TR[1]`):

```
[14] pry(#<RSpec::ExampleGroups::ProductImport::WhenImportingProductsFromUploadedFile>)> page.find(row_containing_name("Carrots"))
=> #<Capybara::Node::Element tag="tr" path="//HTML[1]/BODY[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/TURBO-FRAME[1]/DIV[3]/DIV[1]/FORM[1]/TABLE[1]/TBODY[3]/TR[1]">
```

I think we're aiming to look at the second row (i.e., the first variant row - `TR[2]`). The current implementation looks at the whole section, without specifying the row:

```
[8] pry(#<RSpec::ExampleGroups::ProductImport::WhenImportingProductsFromUploadedFile>)> page.find("#product_#{carrots.id}")
=> #<Capybara::Node::Element tag="tbody" path="//HTML[1]/BODY[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/TURBO-FRAME[1]/DIV[3]/DIV[1]/FORM[1]/TABLE[1]/TBODY[3]">
```

> (You'll need to include ProductsHelper at the top of the file)

I've added it to the base_spec_helper so we don't need to include it; do you think it's better to have it loaded only when needed?

Comment (dacook) [reply spec/system/admin/product_import_spec.rb:689]: > I'm not sure I can make it work with row_containing_name

Oh well, thanks for trying, sorry to have made more work for you.

> I've added it to the base_spec_helper so we don't need to include it; do you think it's better to have it loaded only when needed?

Oh I see. As it's for only a specific part of the system, I personally would have only included it on the relevant tests (I guess I forgot that from a prior review). This is fine too :)

File: spec/system/admin/product_import_spec.rb
Comment (dacook) [reply spec/system/admin/product_import_spec.rb:693]: I find that the biggest challenge is to write simple code ;)

And it's a balance deciding how much time to spend on it! I think that is fine as it is.

**Review by dacook — APPROVED** [review]
This is looking good! I'm not sure if it's ready, but looks good to me so far.

I think the PR description needs updating; I believe it closes #12627 

_edit_: I just saw your final comment and I think you intended it to be ready, so I'll request a second review and update the description.

**Review by rioug — APPROVED** [review]
Looks good :+1: 
I am not really across the plan for BUU testing, but are removing test for the legacy bulk edit product page ? It seems a bit early as it's still in use.

**Comment by dacook** [comment]
> I am not really across the plan for BUU testing, but are removing test for the legacy bulk edit product page ? It seems a bit early as it's still in use.

Oh actually that's a really good point, we don't need to be removing it as part of this PR! I think it should still work. @filipefurtad0 would you agree to restore it?

**Review by dacook — APPROVED** [review]
Great, I think this is ready to go.

However, as there's currently some blocking bugs on the new product screen, I won't merge right now, because it might cause drama for new developers.

**Comment by RachL** [comment]
Scenario 1: All good 🎉 

image


Scenario 2: All good 🎉 

image


In both cases I briefly checked the UI as well. LGTM! Merging

**Pull request merged**
